freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserService.java

```diff
@@ -176,13 +182,44 @@ private SyncStatusDetail synchronizeStack(Stack stack, UsersState umsUsersState,
             removeUsers(freeIpaClient, stateDifference.getUsersToRemove());
             removeGroups(freeIpaClient, stateDifference.getGroupsToRemove());
 
+            // Check for the password related attribute (cdpUserAttr) existence and go for password sync.
+            processUsersWorkloadCredentials(stack.getEnvironmentCrn(), umsUsersState, freeIpaClient);
+
             return SyncStatusDetail.succeed(environmentCrn, "TODO- collect detail info");
         } catch (Exception e) {
             LOGGER.warn("Failed to synchronize environment {}", stack.getEnvironmentCrn(), e);
             return SyncStatusDetail.fail(environmentCrn, e.getLocalizedMessage());
         }
     }
 
+    private void processUsersWorkloadCredentials(String environmentCrn, UsersState umsUsersState, FreeIpaClient freeIpaClient) throws Exception {
+        Config config = freeIpaClient.getConfig();
+        if (config.getIpauserobjectclasses() != null && config.getIpauserobjectclasses().contains(Config.CDP_USER_ATTRIBUTE)) {
```

Comment:
style nit: invert this condition and return early if there is nothing to do. In general it's easier to read code with fewer levels of indentation